File: code/Java/src/main/java/App/MarsRoverEngine.java
package App;

import Commands.ICommand;
import Model.Coordinate;
import Model.Direction;
import Model.Position;

import java.util.ArrayList;

public class MarsRoverEngine {
    private Position position = new Position(0 , 0, "N");

    public void execute(ArrayList<ICommand> commands) {

        for (ICommand command : commands) {
            this.position = command.execute(this.position);
        }
    }

    public Position getPosition() {
        return this.position;
    }
}

File: code/CSharp/ConnascentMarsRover/Source/App/MarsRoverEngine.cs
using System.Collections.Generic;
using Source.Commands;
using Source.Model;

namespace Source.App
{
    public class MarsRoverEngine
    {
        Position _position = new Position(new Coordinate(0,0), Direction.NORTH);
        public void Execute(List<ICommand> commands)
        {
            foreach (var command in commands)
            {
                _position = command.Execute(_position);
            }
        }

        public Position GetPosition()
        {
            return _position;
        }
    }
}